featuregates/client/pkg/featuregateclient/validation.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package featuregateclient
+
+import (
+	"fmt"
+
+	core "github.com/vmware-tanzu/tanzu-framework/apis/core/v1alpha2"
+)
+
+// validateFeatureActivationToggle ensures the given Feature can be activated.
+func validateFeatureActivationToggle(gates *core.FeatureGateList, feature *core.Feature) error {
+	if err := featureExistsInOneAndOnlyOneFeaturegate(gates, feature.Name); err != nil {
+		return fmt.Errorf("validate Feature activation: %w", err)
+	}
+
+	if err := featureActivationToggleAllowed(feature); err != nil {
+		return fmt.Errorf("validate Feature activation: %w", err)
+	}
+
+	return nil
+}
+
+// featureExistsInOneFeaturegate checks that the Feature exists in one and only one FeatureGate.
+func featureExistsInOneAndOnlyOneFeaturegate(gates *core.FeatureGateList, featureName string) error {
+	var n int
+
+	for i := range gates.Items {
+		for _, ref := range gates.Items[i].Spec.Features {
+			if ref.Name == featureName {
+				n++
```

Comment:
can we simply check for `n`'s value here and return?

Reply:
Well, yes, but here's why I didn't do that:
- if I return n, I'm forced to go through and count how many Feature Refs there are. I don't know that if I see more than 1, I can return.
- if I return n, I'll need to check in another function if n is 0 or more than 1. I guess it's separate, but I think it's easier to follow my intention all in one place. But maybe not.

So, the question is, do you think I should check if n is not 1 elsewhere and have a count of a particular Feature in a separate function? I can do that.